### ISSUE TITLE
Khiops distributed execution fails to exchange temporary files

### ISSUE THREAD

**Issue by sgouache** [KhiopsML/khiops#117]
#### Description
When launching khiops in distributed mode over a k8s cluster with mpijob, it fails with a message indicating that a temporary file (file://khiops-runner-worker-0//home/mpiuser/tmp/~Khiops_1/~S1C1_Adult.txt) could not be accessed.

I launched khiops on a master configured with a hostfile listing 2 nodes having 4 cores each like follows:
$ cat /etc/mpi/hostfile 
khiops-runner-worker-0.khiops-runner-worker.dataproc-test.svc:4
khiops-runner-worker-1.khiops-runner-worker.dataproc-test.svc:4

The program runs fine if a single node is used.

#### Questions/Ideas
- the path seems to be influenced by the KHIOPS_TMP_DIR variable
- I guess the URI indicates a temporary file stored on khiops-runner-worker-0, indeed the file can be found there at /home/mpiuser/tmp/~Khiops_1/~S1C1_Adult.txt
- if KHIOPS_TMP_DIR is s3://something the problematic path seems to become s3://something/xxx and program fails with a different error (Unable to open output file)
- the fact that the URI scheme is allowed to be s3 is probably just another distraction but if it's forbidden the program should exit with a clear error message

### Context
- Khiops version 10.2.0
- Log file [khiops.log](https://github.com/KhiopsML/khiops/files/13887725/khiops.log)
- Scenario file [train.kh.txt](https://github.com/KhiopsML/khiops/files/13877791/train.kh.txt)
- OS description Ubuntu 22.04.3 LTS

**Comment by marcboulle** [comment]
### Analysis of traces
The input files in the cloud (s3).
The temps dir is /tmp.
There are two machines: khiops-worker-0 and khiops-worker-1

Log file khiops.log shows that:
- the first train tasks end correctly 
  - File indexer
  - Database basic stats
  - Database slicer
    - write temp files in temp dir
    - should be correct (otherwise, error message in creating or writing these files)
- the folowing task fails
  -  Univariate data preparation
~~~
error : File file://khiops-worker-0//tmp/~Khiops/~S3C1_Adult.txt : Unable to read file (Success)
error : Univariate data preparation : An error occurred in worker while processing (subtask 2)
~~~

Overall, there is no crash and the log reports an error:
- in reading a temporary file in a slave process of task "Univariate data preparation", created in the preceding task "Database slicer"
- the file URI looks correct
- surprisingly, there is an error during an open in read, but the system erreur is 'Success'

### Another issue
Stephane tried to set the temp dir in the cloud (using a s3 uri), and this resulted in a write error in the task "Database slicer". This is "normal", as the temp dir is supposed to be a local path.

Suggestion: add a test with an error message if the temp dir is not a local path

**Comment by bruno-at-orange** [comment]
fixed in merged PR #206
